app/lib/efile/md/md502_calculator.rb

```diff
@@ -255,7 +273,108 @@ def calculate_line_d_count_total
       def calculate_line_d_amount_total
         # Add line A, B and C amounts
         line_or_zero(:MD502_LINE_A_AMOUNT) + line_or_zero(:MD502_LINE_B_AMOUNT) + line_or_zero(:MD502_LINE_C_AMOUNT)
+      end
+
+      def calculate_line_1e
+        total_interest = @direct_file_data.fed_taxable_income + @direct_file_data.fed_tax_exempt_interest
+        total_interest > 11_600
+      end
+
+      def calculate_line_7; end
+
+      def calculate_line_15; end
+
+      def calculate_line_16; end
+
+      FILING_MINIMUMS_NON_SENIOR = {
+        single: 14_600,
+        dependent: 14_600,
+        married_filing_jointly: 29_200,
+        married_filing_separately: 14_600,
+        head_of_household: 21_900,
+        qualifying_widow: 29_200
+      }
+
+      FILING_MINIMUMS_SENIOR = {
+        single: 16_550,
+        dependent: 16_550,
+        married_filing_jointly: 30_750,
+        married_filing_separately: 14_600,
+        head_of_household: 23_850,
+        qualifying_widow: 30_750
+      }
+
+      def calculate_deduction_method
+        gross_income_amount = @intake.tax_calculator.gross_income_amount
+        filing_minimum = if @intake.primary_senior? && @intake.spouse_senior? && @intake.filing_status_mfj?
+                           32_300
+                         elsif @intake.primary_senior?
+                           FILING_MINIMUMS_SENIOR[@intake.filing_status]
+                         else
+                           FILING_MINIMUMS_NON_SENIOR[@intake.filing_status]
+                         end
+        if gross_income_amount >= filing_minimum
+          "S"
+        else
+          "N"
+        end
+      end
+
+      DEDUCTION_TABLES = {
+        s_mfs_d: {
+          12000 => 1_800,
+          17999 => ->(x) { x * 0.15 },
+          18000 => 2_700,
+        },
+        mfj_hoh_qss: {
+          24333 => 3_650,
+          36332 => ->(x) { x * 0.15 },
+          36333 => 5_450,
+        }
+      }.freeze
+      FILING_STATUS_GROUPS = {
+        s_mfs_d: [:single, :married_filing_separately, :dependent],
+        mfj_hoh_qss: [:married_filing_jointly, :head_of_household, :qualifying_widow]
+      }.freeze
+
+      def calculate_line_17
+        if @lines[:MD502_DEDUCTION_METHOD]&.value == "S"
```

Comment:
[dust] would it be too much to define an enum for "S" and "N"? what about defining `standard?` and `non_taxable?` methods? I've seen a recent push against using comparisons like this

Reply:
i'm not sure how to define an enum but we could def add a helper method that takes in the line value